File: go/internal/scoring/testdata/structural-v1/java/Indexed.java
import java.util.HashMap;
import java.util.Map;

final class Indexed {
  static int run(int[] values, int[] queries) {
    Map<Integer, Integer> index = new HashMap<>();
    for (int i = 0; i < values.length; i++) index.put(values[i], i);
    int found = 0;
    for (int query : queries) found += index.getOrDefault(query, 0);
    return found;
  }
}

File: go/internal/scoring/testdata/structural-v1/rust/indexed.rs
use std::collections::HashMap;

pub fn indexed(values: &[i32], queries: &[i32]) -> i32 {
    let mut index = HashMap::new();
    for (i, value) in values.iter().enumerate() { index.insert(*value, i as i32); }
    let mut found = 0;
    for query in queries { found += index.get(query).copied().unwrap_or(0); }
    found
}
